src/llvm/ext/llvm_ext.cc

```diff
@@ -272,6 +272,27 @@ LLVMMetadataRef LLVMExtDIBuilderCreateReplaceableCompositeType(
   return wrap(CT);
 }
 
+LLVMMetadataRef LLVMExtDIBuilderCreateUnspecifiedType(
+  DIBuilderRef Dref, const char *Name, size_t NameLen) {
+#if LLVM_VERSION_GE(7, 0)
+  return LLVMDIBuilderCreateUnspecifiedType(wrap(Dref), Name, NameLen);
+#else
+  return wrap(Dref->createUnspecifiedType({Name, NameLen}));
+#endif
+}
+
+LLVMMetadataRef LLVMExtDIBuilderCreateLexicalBlockFile(
+  DIBuilderRef Dref,
+  LLVMMetadataRef Scope, LLVMMetadataRef File, unsigned Discriminator) {
+#if LLVM_VERSION_GE(7, 0)
+  return LLVMDIBuilderCreateLexicalBlockFile(wrap(Dref), Scope, File, Discriminator);
+#else
+  return wrap(Dref->createLexicalBlockFile(unwrapDI<DIScope>(Scope),
+                                           unwrapDI<DIFile>(File),
+                                           Discriminator));
+#endif
+}
```

Comment:
```suggestion
LLVMMetadataRef LLVMExtDIBuilderCreateUnspecifiedType(
  DIBuilderRef Dref, const char *Name, size_t NameLen) {
  return wrap(Dref->createUnspecifiedType({Name, NameLen}));
}

LLVMMetadataRef LLVMExtDIBuilderCreateLexicalBlockFile(
  DIBuilderRef Dref,
  LLVMMetadataRef Scope, LLVMMetadataRef File, unsigned Discriminator) {
  return wrap(Dref->createLexicalBlockFile(unwrapDI<DIScope>(Scope),
                                           unwrapDI<DIFile>(File),
                                           Discriminator));
}
```

I think it's best to just implement the function directly, without conditionals, and add a TODO to use LLVM instead when older support is dropped.

This is the only thing that helped in my case.
https://github.com/oprypin/crystal/runs/597786291#step:18:109

The main thing that might be causing problems for me is that this introduces a dependency on LLVM-C in llvm_ext?